var/spack/repos/builtin/packages/qthreads/package.py

```diff
@@ -85,3 +85,58 @@ def configure_args(self):
                     % self.spec.variants['scheduler'].value)
 
         return args
+
+
+
+    test_base_path = 'test/basics/'
+    test_list = ['hello_world_multi', 'hello_world']
+
+    @run_after('install')
+    def setup_build_tests(self):
+        """Copy the build test files after the package is installed to an
+        install test subdirectory for use during `spack test run`."""
+        tests = self.test_list
+        relative_test_dir = self.test_base_path
+
+        tests_to_cpy = []
+        header = 'test/argparsing.h'
+        for test in tests:
+            test_path = join_path(relative_test_dir, test + '.c')
+            tests_to_cpy.append(test_path)
+        
+        self.cache_extra_test_sources(tests_to_cpy)
+        self.cache_extra_test_sources(header) 
+
+    def build_tests(self):
+        """Build and run the added smoke (install) test."""
+        tests = self.test_list
+        relative_test_dir = self.test_base_path
+
+        for test in tests:
+            options = [
+               '-I{0}'.format(self.prefix.include),
```

Comment:
FWIW. I had to add `'-std=c11'` as an option to get this to compile for gcc@8.1.0 on a linux box and actually run the tests.

As I mentioned in email, I get more focused error message in the test output with the default scheduler *and* the all of the tests pass for me when the library is built with the `distrib` scheduler.  (Thanks for pointing me to it!)